File: 3095-shortest-subarray-with-or-at-least-k-i/3095-shortest-subarray-with-or-at-least-k-i.java
class Solution {
    public int minimumSubarrayLength(int[] nums, int k) {
        
        int n = nums.length;
        int ans = Integer.MAX_VALUE;

        for (int i = 0; i < n; i++) {
            int curOr = 0;
            for (int j = i; j < n; j++) {
                curOr |= nums[j];
                if (curOr >= k) {
                    ans = Math.min(ans, j - i + 1);
                    break;
                }
            }
        }
        return ans == Integer.MAX_VALUE ? -1 : ans;
    }
}

File: 3095-shortest-subarray-with-or-at-least-k-i/3095-shortest-subarray-with-or-at-least-k-i.py
class Solution:
    def minimumSubarrayLength(self, nums: List[int], k: int) -> int:
        

        n = len(nums)
        INF = 10 ** 10
        best = INF
        for i in range(n):
            for j in range(i, n):
                sub = nums[i:j + 1]
                print(sub)
                bit_sum = 0
                for x in sub:
                    bit_sum |= x
                
                if bit_sum >= k:
                    best = min(best, len(sub))
        
        return best if best != INF else -1

